src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseStudentParticipationRepository.java

```diff
@@ -91,6 +91,18 @@ default ProgrammingExerciseStudentParticipation findWithSubmissionsByExerciseIdA
 
     Optional<ProgrammingExerciseStudentParticipation> findByExerciseIdAndStudentLoginAndTestRun(long exerciseId, String username, boolean testRun);
 
+    @Query("""
+            SELECT participation
+            FROM ProgrammingExerciseStudentParticipation participation
+            WHERE participation.exercise.id = :exerciseId
+                AND participation.student.login = :username
+                AND participation.testRun = :testRun
+            ORDER BY participation.id DESC
+            LIMIT 1
```

Comment:
`LIMIT` is **not** a generally supported keyword and will break Artemis on Postgres.